tool/tsh/tsh.go

```diff
@@ -1259,9 +1264,16 @@ func newTraceProvider(cf *CLIConf, command string, ignored []string) (*tracing.P
 
 // onVersion prints version info.
 func onVersion(cf *CLIConf) error {
-	proxyVersion, err := fetchProxyVersion(cf)
-	if err != nil {
-		fmt.Fprintf(cf.Stderr(), "Failed to fetch proxy version: %s\n", err)
+	proxyVersion := ""
+	proxyPublicAddr := ""
+	// Do not check proxy version if client only version parameter is given
+	if !cf.clientOnlyVersionCheck {
+		pv, ppa, err := fetchProxyVersion(cf)
```

Comment:
Should we put a timeout on this? If the proxy is not reachable, how long does this take to time out?

Reply:
It's fairly instant, `0.233` in this case.  Example here.  Dont' see a need currently. 
```bash
$ time tsh version
Failed to fetch proxy version: Get "https://teleport.example.com:443/webapi/ping/local": dial tcp 1.2.3.4:443: connect: connection refused
Teleport v13.0.0-dev git:v12.0.0-dev.hatch.ami.1-1212-g1887217811 go1.19.5
tsh version  0.04s user 0.07s system 46% cpu 0.233 total
```

Reply:
Well, it's fairly instant in the case where you get a TCP RST packet back to tell you that nothing's listening on that port, but if the port it blocked by a firewall (for example) it will just hang.

A version command is expected to run very fast, so I think a timeout is important here.

Reply:
There is a 5 second timeout in `fetchProxyVersion`
```go
ctx, cancel := context.WithTimeout(cf.Context, time.Second*5)
```